### PR TITLE
Add diagnostics endpoint

### PR DESCRIPTION
**Pull request by iMicknl**
Decided to wrap it in a new function call and not extend `get_setup()` since that will mess up the typing a lot, and now we can directly mask the data.

```
        setup = await client.get_diagnostic_data()
```

### CODE REVIEW

File: pyoverkiz/utils.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+"""Utils for Overkiz client"""
```
Comment (tetienne) [inline pyoverkiz/utils.py:1]: Not a big fan of `utils` name. This is too generic. I would suggest something like `obfuscate.py` or `mask.py`.

File: pyoverkiz/utils.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    """Mask id"""
+    return re.sub(r"(SETUP)?\d+-", "****-", str(id))
+
+
+def obfuscate_email(email: str | None) -> str:
+    """Mask email"""
+    return re.sub(r"(.).*@.*(.\..*)", r"\1****@****\2", str(email))
+
+
+def mask(input: str) -> str:
```
Comment (tetienne) [inline pyoverkiz/utils.py:19]: `mask` vs `obfuscate`, we have to choose.

Comment (iMicknl) [reply pyoverkiz/utils.py:19]: For now I just ported the names from our current `models.py`. However happy to rename and list as breaking change.

File: pyoverkiz/utils.py (hunk 1 above)
Comment (iMicknl) [reply pyoverkiz/utils.py:1]: Fine by me as well, let's change it to what we decide the function name to be.

File: pyoverkiz/utils.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    """Mask string"""
+    return re.sub(r"[a-zA-Z0-9_.-]*", "*", str(input))
+
+
+# pylint: disable=too-many-nested-blocks
+def mask_json(data: dict) -> JSON:
```
Comment (tetienne) [inline pyoverkiz/utils.py:25]:
```suggestion
def mask_sensitive_data(data: dict) -> JSON:
```

Comment (tetienne) [reply pyoverkiz/utils.py:25]: dict is also missing type.

File: pyoverkiz/utils.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        if isinstance(value, dict):
+            mask_json(value)
+        elif isinstance(value, list):
+            for val in value:
+                if isinstance(val, str):
+                    pass
+                elif isinstance(val, list):
+                    pass
+                else:
+                    mask_json(val)
```
Comment (tetienne) [inline pyoverkiz/utils.py:62]:
```suggestion
                if isinstance(val, str):
                    continue
                elif isinstance(val, list):
                    continue
                else:
                    mask_json(val)
```

File: pyoverkiz/client.py [hunk 5]
```diff
@@ -272,6 +273,27 @@ async def get_setup(self, refresh: bool = False) -> Setup:
...
+            -> setup location: <location>
+            -> house places (rooms and floors): <place>
+            -> setup devices: <devices>
+
+        This data will be masked to not return any confidential or PII data.
+        """
+        response = await self.__get("setup")
+        setup = obfuscate_sensitive_data(response)
+
+        return setup
```
Comment (tetienne) [inline pyoverkiz/client.py:295]:
```suggestion

        return obfuscate_sensitive_data(response)
```

**Review by tetienne — APPROVED** [review]

**Pull request merged**